packages/flutter/lib/src/widgets/ticker_provider.dart

```diff
@@ -0,0 +1,210 @@
+/// [TickerProviderStateMixin] instead.
+abstract class SingleTickerProviderStateMixin implements State<dynamic>, TickerProvider { // ignore: TYPE_ARGUMENT_NOT_MATCHING_BOUNDS, https://github.com/dart-lang/sdk/issues/25232
+
+  Ticker _ticker;
+
+  @override
+  Ticker createTicker(TickerCallback onTick) {
+    assert(() {
+      if (_ticker != null) {
+        throw new FlutterError(
+          '$runtimeType is a SingleTickerProviderStateMixin but multiple tickers were created.\n'
+          'A SingleTickerProviderStateMixin can only be used as a TickerProvider once. If a '
+          'State is used for multiple AnimationController objects, or if it is passed to other '
+          'objects and those objects might use it more than one time in total, then instead of '
+          'mixing in a SingleTickerProviderStateMixin, use a regular TickerProviderStateMixin.'
+        );
+      }
+      return true;
+    });
+    _ticker = new Ticker(onTick, debugLabel: 'created by $this');
+    return _ticker;
+  }
+
+  @override
+  void dispose() {
+    assert(() {
+      if (_ticker == null || !_ticker.isActive)
+        return true;
+      throw new FlutterError(
+        '$this was disposed with an active Ticker.\n'
+        '$runtimeType created a Ticker via its SingleTickerProviderStateMixin, but at the time '
+        'dispose() was called on the mixin, that Ticker was still active. The Ticker must '
+        'be disposed before calling super.dispose(). Tickers used by AnimationControllers '
+        'should be disposed by calling dispose() on the AnimationController itself. '
+        'Otherwise, the ticker will leak.\n'
+        'The offending ticker was: ${_ticker.toString(debugIncludeStack: true)}'
+      );
+    });
+    super.dispose();
+  }
+
+  @override
+  void dependenciesChanged() {
+    final bool muted = !TickerMode.of(context);
+    _ticker.muted = muted;
+    super.dependenciesChanged();
+  }
+
+  @override
+  void debugFillDescription(List<String> description) {
+    super.debugFillDescription(description);
+    if (_ticker != null) {
+      if (_ticker.isActive && _ticker.muted)
+        description.add('ticker active but muted');
+      else
+      if (_ticker.isActive)
+        description.add('ticker active');
+      else
+      if (_ticker.muted)
+        description.add('ticker inactive and muted');
+      else
+        description.add('ticker inactive');
+    }
+  }
+
+}
+
+/// Provides [Ticker] objects that are configured to only tick while the current
+/// tree is enabled, as defined by [TickerMode].
+///
+/// To create an [AnimationController] in a class that uses this mixin, pass
+/// `tickerProvider: this` to the animation controller constructor whenever you
+/// create a new animation controller.
+///
+/// If you only have a single [Ticker] (for example only a single
+/// [AnimationController]) for the lifetime of your [State], then using a
+/// [SingleTickerProviderStateMixin] is more efficient. This is the common case.
```

Comment:
I guess 'That' in this particular case underlines the separation of the two ideas. To me, 'This' brings the two closer than they naturally are, but this may be personal bias.